src/machine learning analysis.py

```diff
@@ -0,0 +1,155 @@
+# author: Siqi Tao
+# date: 2021-11-26
+
+"""Downloads data as a csv file from online to a local filepath via a URL.
+Usage: download_script.py --url=<url> --filepath=<filepath> 
+ 
+Options:
+--input=<input>       The path or filename pointing to the data
+--output=<output>     The prefix where to write the output figure(s)/table(s) to and what to call it 
+"""
+
+import os
+import pandas as pd
+from docopt import docopt
+
+import numpy as np
+import pandas as pd
+from sklearn.compose import ColumnTransformer, make_column_transformer
+from sklearn.linear_model import LogisticRegression
+from sklearn.metrics import make_scorer
+from sklearn.model_selection import (
+    RandomizedSearchCV,
+    cross_val_score,
+    cross_validate
+)
+from sklearn.pipeline import Pipeline, make_pipeline
+from sklearn.preprocessing import OneHotEncoder
+from scipy.stats import loguniform
+from sklearn.metrics import confusion_matrix
+import pickle
+
+opt = docopt(__doc__)
+
+def main(): 
+    
+    # Get data
+    input = opt["--input"]
```

Comment:
docopt parameters should be passed through the function call.

`def main(input)` where input is being passed `opt["--input"]` from the function call at the very end.

Reply:
it seems unnecessary to pass the parameters in python script?
https://stackoverflow.com/questions/53204814/is-it-bad-practice-to-have-arguments-called-in-main-in-python

Reply:
The context matters I suppose. Since we are using docopt, you should pass docopt arguments through the function call. You are anyway assigning it to the `input` variable inside the `main()` function which is redundant if you pass it through the function call. If it were a regular script where we have global variables defined, then we can directly reference them in the code as show in the link above.